ruler-frontend/src/main/kotlin/com/spotify/ruler/frontend/components/Insights.kt

```diff
@@ -33,30 +36,30 @@ import react.dom.p
 import react.useEffect
 
 @RFunction
-fun RBuilder.insights(components: List<AppComponent>) {
+fun RBuilder.insights(insights: Insights) {
     div(classes = "row mb-3") {
-        fileTypeGraphs(components)
+        fileTypeGraphs(insights.fileTypeInsights)
     }
     div(classes = "row") {
-        componentTypeGraphs(components)
+        componentTypeGraphs(insights.componentTypeInsights)
     }
     div(classes = "row") {
-        resourcesTypeGraphs(components)
+        resourcesTypeGraphs(insights.resourcesTypeInsights)
     }
 }
 
 @RFunction
-fun RBuilder.fileTypeGraphs(components: List<AppComponent>) {
-    val labels = arrayOf("Classes", "Resources", "Assets", "Native libraries", "Other")
+fun RBuilder.fileTypeGraphs(fileTypeInsights: Map<FileType, TypeInsights>) {
+    val labels = FileType.values().map { it.label }.toTypedArray()
     val downloadSizes = LongArray(labels.size)
     val installSizes = LongArray(labels.size)
     val fileCounts = LongArray(labels.size)
 
-    components.flatMap(AppComponent::files).forEach { file ->
-        val index = file.type.ordinal
-        downloadSizes[index] += file.getSize(Measurable.SizeType.DOWNLOAD)
-        installSizes[index] += file.getSize(Measurable.SizeType.INSTALL)
-        fileCounts[index]++
+    fileTypeInsights.forEach { (fileType, insights) ->
+        val index = fileType.ordinal
+        downloadSizes[index] = insights.downloadSize
+        installSizes[index] = insights.installSize
+        fileCounts[index] = insights.count
```

Comment:
insights are already calculated, here we are just populating the arrays passed to the frontend. Same for the two bellow methods